xc/xenops_server_xen.ml

```diff
@@ -1555,6 +1555,12 @@ module VM = struct
 					)
 			) Oldest task vm
 
+	let inject_igmp_query domid vifs =
+		let vif_names = List.map (fun vif -> Printf.sprintf "vif%d.%d" domid vif.Vif.position) vifs in
+		debug "Inject IGMP query to %s" (String.concat " " vif_names);
+		(* Call script to inject IGMP query asynchronously *)
+		Forkhelpers.execute_command_get_output !Xc_resources.igmp_query_injector_script (["--detach"; "vif"; "--wait-vif-connected"; string_of_int !Xenopsd.vif_ready_timeout] @ vif_names)
```

Comment:
Consider writing the list like `"--detach" :: "vif" :: ... :: vif_names`, which is a little more efficient.